xarray/tests/test_groupby.py

```diff
@@ -3147,7 +3147,36 @@ def test_time_grouping_seasons_specified():
     assert_identical(actual, expected.reindex(season=labels))
 
 
-def test_groupby_multiple_bin_grouper_missing_groups():
+def test_multiple_grouper_unsorted_order() -> None:
```

Comment:
The code is hard to review, but I would appreciate someone taking a look at this test.